src/Azure.Functions.Cli/Kubernetes/KEDA/KedaHelper.cs

```diff
@@ -41,10 +43,20 @@ internal static IKubernetesResource GetScaledObject(string name, string @namespa
             }
         }
 
-        public static KedaVersion DetermineCurrentVersion(string @namespace)
+        public static async Task<KedaVersion> DetermineCurrentVersion(string @namespace)
         {
-            // TODO: discover version
-            return KedaVersion.v2;
+            // Get KEDA resource information
+            var resourceInfo = await KubernetesHelper.ResourceExists("Deployment", "keda-metrics-apiserver", @namespace, returnJsonOutput: true);
+            if (resourceInfo.ResourceExists == false)
+            {
+                // We use KEDA v2 as a default
+                return KedaVersion.v2;
+            }
```

Comment:
Perhaps checking for the existence of the apiservice here instead might be more compatible with different methods of installing helm?

```suggestion
            var resourceInfoV2 = await KubernetesHelper.ResourceExists("apiservice", "v1alpha1.keda.sh", @namespace, returnJsonOutput: true);
            
            if(resourceInfoV2.ResourceExists)
            {
                return KedaVersion.v2;
            }
```

Reply:
This way you rely on the resource type which is different between v1 & v2 which is not the case.

The ways to install KEDA all provide deployment with that name so think we are good to go here.